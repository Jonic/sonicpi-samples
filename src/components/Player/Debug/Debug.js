import './Debug.scss'

import React, { useContext } from 'react'

import PlayerContext from 'contexts/Player'

const Debug = () => {
  const { currentCategory, currentSample, isLooping, isPlaying } = useContext(
    PlayerContext,
  )

  return (
    <div className="c-debug">
      <ul className="c-debug__list">
        <li className="c-debug__list-item">
          currentCategory: {currentCategory}
        </li>
        <li className="c-debug__list-item">currentSample: {currentSample}</li>
        <li className="c-debug__list-item">
          isLooping: {isLooping.toString()}
        </li>
        <li className="c-debug__list-item">
          isPlaying: {isPlaying.toString()}
        </li>
      </ul>
    </div>
  )
}

export default Debug
