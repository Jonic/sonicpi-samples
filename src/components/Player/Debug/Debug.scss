@import '../../../styles/component-helper';

.c-debug {
}

.c-debug__list {
}

.c-debug__list-item {
}
