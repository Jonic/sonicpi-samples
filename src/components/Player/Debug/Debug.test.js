import Debug from './Debug'

const currentCategory = ''
const currentSample = ''
const isLooping = false
const isPlaying = false

let wrapper

describe('<Debug />', () => {
  beforeEach(() => {
    wrapper = shallow(
      <Debug
        currentCategory={currentCategory}
        currentSample={currentSample}
        isLooping={isLooping}
        isPlaying={isPlaying}
      />,
    )
  })

  describe('Renders...', () => {
    it('without crashing', () => {
      expect(wrapper.find('.c-debug')).toExist()
    })

    it('matches its snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    })
  })
})
