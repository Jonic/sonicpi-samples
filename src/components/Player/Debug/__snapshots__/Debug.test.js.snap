// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Debug /> Renders matches its snapshot 1`] = `
Array [
  <Debug>
    <div
      className="c-debug"
    >
      <ul
        className="c-debug__list"
      >
        <li
          className="c-debug__list-item"
        >
          currentCategory: 
        </li>
        <li
          className="c-debug__list-item"
        >
          currentSample: 
        </li>
        <li
          className="c-debug__list-item"
        >
          isLooping: 
          false
        </li>
        <li
          className="c-debug__list-item"
        >
          isPlaying: 
          true
        </li>
      </ul>
    </div>
  </Debug>,
  ",",
]
`;
